### DISCUSSION TITLE
The package javafx.beans.value is accessible from more than one module: <unnamed>, javafx.base

### DISCUSSION THREAD

**Discussion by chgenly** [eclipse-tycho/tycho#2219] [Q&A] [answered]
During a build I get the following error.

     [exec] [ERROR] C:\coherent\tot\src\hxisde\com.coherentlogix.rta.logger\src\com\coherentlogix\rta\logger\LoggerPart.java:[58]
     [exec] [ERROR]     import javafx.beans.value.ChangeListener;
     [exec] [ERROR]            ^^^^^^^^^^^^^^^^^^
     [exec] [ERROR] The package javafx.beans.value is accessible from more than one module: <unnamed>, javafx.base

I think the unnamed package is a problem.   My question is, how do I figure out why it is attempting to access the unnamed module?  Are there debugging options?

Interestingly, this problem occurred when I tried to move from Java 11 to Java 17.
<project ...
  <properties>
       <maven.compiler.source>17</maven.compiler.source>
      <maven.compiler.target>17</maven.compiler.target>

**Comment by laeubi** [comment]
Have you made sure that `javafx.beans.value` is properly imported in the manifest and the bundle is at `JavaSE-17`?

**Comment by chgenly** [comment]
There is no mention of javafx.beans.value in the manifest.   The following is included in the manifest.
Bundle-RequiredExecutionEnvironment: JavaSE-17

It builds inside Eclipse just fine.  It's only a problem when I build on the command line using maven.
The javafx.beans.value package in in the module javafx.base which is part of our JRE system library.

**Comment by laeubi** [comment, marked as answer]
> There is no mention of javafx.beans.value in the manifest. The following is included in the manifest.
> Bundle-RequiredExecutionEnvironment: JavaSE-17

Then this is wrong

> It builds inside Eclipse just fine. 

PDE is buggy in this regard because it assumes all packages from a JVM are always imported by default what is no longer true since Java 9+ (and it was even wrong before Java 9).

The rule is simple:

- `java.*` ---> `Bundle-RequiredExecutionEnvironment`
- everything else ---> `Import-Package`

**Reply by chgenly** [reply]
I added
`  Import-Package: javafx.beans.value`
and still have the same result:
 **The package javafx.beans.value is accessible from more than one module: <unnamed>, javafx.base**
It doesn't seem to have trouble finding the module.  The trouble is it finds two modules.  How do I figure out why it is attempting to use the unnamed module?  

And as I said, this worked with Java 11, but it fails with Java 17.  Modules were added in Java 9 (I think).  So I'm surprised it changes from 11 to 17.

**Reply by laeubi** [reply]
The best would be to provide a small example, usually there is one package *not* imported (not necessarily the one there is complains about) and a jar providing that package in the target platform and then you getting conflicts.

**Reply by chgenly** [reply]
The error message in my previous post is incorrect. It removed the unnamed module.
**The package javafx.beans.value is accessible from more than one module: &lt;unnamed>, javafx.base**

**Reply by chgenly** [reply]
Ah! A small example.  That's a very good idea.

**Comment by chgenly** [comment]
A small example was the right thing to work on.

The problem was caused by the parent pom, configuring tycho to use extraClassPathElements.  These elements pointed to outdated javafx versions.  And in fact these elements could be omitted altogether.

I very much appreciate your help.
